### PR TITLE
Fix energy loss and range tables

### PR DESCRIPTION
**Pull request by amandalund**
This is a temporary fix for #253. Previously we were importing a mix of per-process dE/dx tables, total dE/dx summed over processes, and total range but treating them all like per-process dE/dx and range. With this change only the summed tables are loaded, and the physics stepping is modified to use total energy loss and range. We might eventually want to refactor the value grids so the summed dE/dx and range tables are stored in a separate data structure rather than using the same awkward approach as Geant4 where they belong to an arbitrary energy loss process.

There is one final discrepancy between the updated geant-exporter-data.root and the original data which @stognini and I are still trying to clear up.

### CODE REVIEW

File: src/physics/base/Process.hh [hunk 1]
```diff
@@ -19,6 +19,17 @@ namespace celeritas
...
 
+//---------------------------------------------------------------------------//
+/*!
+ * Type of physics process.
+ */
+enum class ProcessType
+{
+    electromagnetic_discrete, //!< Discrete EM process
+    electromagnetic_dedx,     //!< Continuous-discrete EM process
+    electromagnetic_msc       //!> Multiple scattering
```
Comment (stognini) [inline src/physics/base/Process.hh:30]: Looks like there was typo here. `!>` instead of `!<`.

Comment (amandalund) [reply src/physics/base/Process.hh:30]: haha oops, good catch!

**Comment by stognini** [comment]
Just to keep this information documented here:

> There is one final discrepancy between the updated geant-exporter-data.root and the original data which @stognini and I are still trying to clear up.

After a bunch of tests and comparisons, @amandalund figured out that the discrepancy we were seeing in the `geant-exporter-data.root` was coming from different CLHEP units (internal G4 CLHEP vs. Spack's).

**Comment by sethrj** [comment]
> After a bunch of tests and comparisons, @amandalund figured out that the discrepancy we were seeing in the `geant-exporter-data.root` was coming from different CLHEP units (internal G4 CLHEP vs. Spack's).

Nice find! How on earth did the units change?!

**Review by sethrj — APPROVED** [review]
Thanks @amandalund , this looks like a good workaround. I'm not a huge fan of the new `type` for each `Process`; I think MSC will end up with a different class interface.

From a runtime perspective I think we should just drop the de/dx and range tables from the individual processes, since only one process per particle type will have them. Instead we could just add `ValueTable` entry for each of the energy loss and range.

One problem that it seems we're having is that I originally set up the data as though our physics processes are separable (where each process *does* have an individual de/dx), but it's awkward to translate the preprocessed Geant4 data through that interface (and indeed the range/inverse range calculation doesn't make sense with the processes separated).

So maybe what we need is a separate `EnergyLossProcess` class, which in the "ideal" case where we calculate all of our cross sections and table a priori can be constructed by compositing a bunch of `Process` classes (which can actually emit de/dx); and in our current "realistic" case can just return the pre-integrated values that we're getting from Geant4.

Anyway, I think this is good for now, but I think that we'll end up deleting the `type` method from the Process and adding a new helper class that gets passed to the physics input.

Let me know which of this or #251 you want to merge first.

**Comment by amandalund** [comment]
Thanks @sethrj, I like that solution and agree about the `type` method. I'd say merge this one before the demo loop updates.

**Pull request merged**